crates/biome_js_analyze/src/utils.rs

```diff
@@ -95,3 +95,120 @@ pub(crate) fn is_node_equal(a_node: &JsSyntaxNode, b_node: &JsSyntaxNode) -> boo
     }
     true
 }
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum VariablePosition {
+    Right,
+    Left,
+}
+
+/// Finds the position of a variable relative to a binary expression.
+///
+/// This function takes a reference to a JsBinaryExpression and the name of a variable as input.
+/// It determines whether the variable appears to the left or right of the binary operator within
+/// the expression and returns the result as an `Option<VariablePosition>`.
+///
+/// Depending on your specific expression and variable placement,
+/// the result may vary between Left, Right, or None.
+pub(crate) fn find_variable_position(
+    binary_expression: &JsBinaryExpression,
+    variable: &str,
+) -> Option<VariablePosition> {
+    let operator_range = binary_expression
+        .operator_token()
+        .ok()?
+        .text_trimmed_range();
+
+    binary_expression
+        .syntax()
+        .children()
+        .filter_map(AnyJsExpression::cast)
+        .map(|child| child.omit_parentheses())
+        .filter(|child| child.syntax().text_trimmed() == variable)
+        .find_map(|child| {
+            if child.syntax().text_trimmed_range().end() < operator_range.start() {
+                Some(VariablePosition::Left)
+            } else if operator_range.end() < child.syntax().text_trimmed_range().start() {
+                Some(VariablePosition::Right)
+            } else {
+                None
+            }
+        })
+}
```

Comment:
That's great @victor-teles !! You took my feedback and did exactly what suggested! 💪

I still think that the branch at lines `133..135` shouldn't exist. Maybe let's try to use `unreachable!("The node can't have the same range of the operator.")` instead of `None`. Theoretically we should never pass a node that is the same range as the operator, and it is we should panic because the developer did something wrong.